beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1407,12 +1385,21 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
     ///
     /// Returns `(block_root, block_slot)`.
     pub fn heads(&self) -> Vec<(Hash256, Slot)> {
-        self.head_tracker.heads()
+        let head_slot = self.canonical_head.cached_head().head_slot();
+        self.canonical_head
+            .fork_choice_read_lock()
+            .proto_array()
+            .viable_heads::<T::EthSpec>(head_slot)
+            .iter()
+            .map(|node| (node.root, node.slot))
+            .collect()
```

Comment:
Could use to implement the DB downgrade (recreate the headtracker after it has been deleted)